fe/fe-core/src/main/java/com/starrocks/catalog/TableProperty.java

```diff
@@ -885,8 +885,14 @@ public void clearBinlogAvailableVersion() {
     }
 
     public TableProperty buildUseFastSchemaEvolution() {
+        String defaultValue;
+        if (RunMode.isSharedDataMode()) {
+            defaultValue = Config.enable_fast_schema_evolution ? "true" : "false";
+        } else {
+            defaultValue = "false";
```

Comment:
The purpose here is to allow fast schema evolution to support tables created in older versions in shared data mode, which is not possible in shared nothing mode because tables created in older versions do not have column unique ids in shared nothing mode

Reply:
this will be a painpoint as the inconsistent default behavior between shared-data and shared-nothing if the table property is not specifically provided with ("use_fast_schema_evolution").

I thought the behavior is, if `enable_fast_schema_evolution = false`. no matter the property is given true or false, the fast schema evolution should be disabled. if turned on, then should obey the property settings or fallback to a default value.

Otherwise, the configvar should be renamed to be shared-data specific.

Reply:
Updated, I've added another config `enable_fast_schema_evolution_in_share_data_mode` to control whether turn this optimization on or not.